File: DbNetSuiteCore/Models/DbNetFile/DbNetFileRequest.cs
using DbNetSuiteCore.Enums;
using System.Collections.Generic;

namespace DbNetSuiteCore.Models.DbNetFile
{
    public class DbNetFileRequest : DbNetSuiteRequest
    {
        public string Folder { get; set; }
        public string FileName { get; set; }
        public string RootFolder { get; set; }
        public List<FileColumn> Columns { get; set; } = new List<FileColumn>();
        public bool QuickSearch { get; set; }
        public ToolbarButtonStyle ToolbarButtonStyle { get; set; }
        public bool Search { get; set; }
        public bool Navigation { get; set; }
        public bool Export { get; set; }
        public bool Copy { get; set; }
        public bool Upload { get; set; }
        public int CurrentPage { get; set; }
        public string Caption { get; set; }
        public bool Nested { get; set; }
        public string OrderBy { get; set; }
        public OrderByDirection? OrderByDirection { get; set; }
        public List<SearchParameter> SearchParams { get; set; }
        public string SearchFilterJoin { get; set; } = "and";
        public bool IsSearchResults { get; set; }
        public bool IncludeSubfolders { get; set; }

    }
}

File: DbNetSuiteCore/Resources/TypeScript/Interfaces/DbNetFile/DbNetFileRequest.ts
interface DbNetFileRequest extends DbNetSuiteRequest
{
    rootFolder: string;
    folder: string;
    columns: FileColumnRequest[];
    quickSearch: boolean;
    quickSearchToken: string;
    toolbarButtonStyle: ToolbarButtonStyle;
    search: boolean;
    searchFilterJoin: string;
    searchParams: Array<SearchParam>;
    export: boolean;
    copy: boolean;
    upload: boolean;
    navigation: boolean;
    pageSize: number;
    currentPage: number;
    caption: string;
    nested: boolean;
    fileName: string;
    orderBy: string;
    orderByDirection: string;
    isSearchResults: boolean;
    includeSubfolders: boolean;
    treeView: boolean;
    filesOnly: boolean;
}